src/core/mavsdk_impl.cpp

```diff
@@ -273,6 +273,7 @@ void MavsdkImpl::set_configuration(Mavsdk::Configuration configuration)
 {
     own_address.system_id = configuration.get_system_id();
     own_address.component_id = configuration.get_component_id();
+    _configuration.set_usage_type(configuration.get_usage_type());
```

Comment:
I don't think this is required, right? Or am I confused?

Reply:
I think it is required, in order to copy the usage type from the Configuration object into the MavsdkImpl object, where it is needed by get_mav_type().

Reply:
Ok.